Client/U2FExtensions.swift

```diff
@@ -613,46 +648,74 @@ class U2FExtensions: NSObject {
             return
         }
         
-        if tab?.id == currentTabId && (fido2Service.keyState == .touchKey || u2fService.keyState == .YKFKeyU2FServiceKeyStateTouchKey) {
-            popup.showWithType(showType: .flyUp)
+        // The modal should be visible for the tab where the U2F API is active
+        if u2fActive && tab?.id == currentTabId && (fido2Service.keyState == .touchKey || u2fService.keyState == .YKFKeyU2FServiceKeyStateTouchKey) {
+            let currentURL = self.tab?.url?.host ?? ""
+            touchKeyPopup.updateTitle(title: Strings.keyTitle + currentURL)
+            touchKeyPopup.showWithType(showType: .flyUp)
             return
         }
-        popup.dismissWithType(dismissType: .flyDown)
+        touchKeyPopup.dismissWithType(dismissType: .flyDown)
     }
     
-    private func handlePinVerificationRequired(completion: @escaping (Bool) -> Void ) {
+    private func handlePinVerificationRequired(completion: @escaping (Bool) -> Void) {
         ensureMainThread {
-            let alert = UIAlertController.userTextInputAlert(title: Strings.pinTitle, message: Strings.pinLabel, placeholder: Strings.pinPlaceholder) {
-                pin, _ in
-                if let pin = pin, !pin.isEmpty {
-                    guard let fido2Service = YubiKitManager.shared.keySession.fido2Service else {
-                        completion(true)
-                        return
-                    }
-                    guard let verifyPinRequest = YKFKeyFIDO2VerifyPinRequest(pin: pin) else {
-                        completion(true)
-                        return
-                    }
-
-                    fido2Service.execute(verifyPinRequest) { (error) in
-                        guard error == nil else {
-                            completion(true)
-                            return
-                        }
-                        completion(false)
-                        return
-                    }
-                } else {
+            self.pinVerificationPopup.addButton(title: Strings.confirmPin) { [weak self] in
+                guard let self = self else {
+                    return .flyDown
+                }
+                return self.verifyPin(completion: completion)
+            }
+            self.pinVerificationPopup.showWithType(showType: .flyUp)
+        }
+    }
+    
+    private func verifyPin(completion: @escaping (Bool) -> Void) -> PopupViewDismissType {
+        if let pin = pinVerificationPopup.text, !pin.isEmpty {
```

Comment:
We should probably `guard` these too to keep this method flatter. 

Also, since so many of these use `completion(true); return;` we may as well just combine the guard statements:
```
guard
    let pin = 
    let fido2Service = 
    let verifyPinRequest =
    else {
        completion(true)
        return
    }
}
```